Add docs about upgrading gitlab and jenkins

ivanchimeno opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
The current documentation does not state how to upgrade a Jenkins instance. Information on upgrading Gitlab to a major version is also missing

### Description
<!-- Describe your changes in detail -->
This PR adds a new section which documents how to upgrade a Jenkins instance via Docker. It also adds additional information on how to conduct Gitlab upgrades to major versions.

b-fein reviewed (commented):
Found a few typos and left a comment regarding the process when using the `docker-compose` quickstart dev-setup.

NicolasUllmann reviewed (approved):
Except for the typos already mentioned above, it looks good to me.

ivanchimeno commented on docs/dev/setup/jenkins-gitlab.rst:
```diff
@@ -922,6 +924,51 @@ the following steps:
+docker image can be found on `dockerhub <https://hub.docker.com/r/jenkins/jenkins/tags?page=1&ordering=last_updated>`__.
+
+1. Open the Jenkins Dockerfile and replace ``jenkins/jenkins:lts`` with the version you want to use.
+   For example, if you want to upgrade Jenkins to version ``2.289.2``, you will need to use the ``jenkins/jenkins:2.289.2-lts`` image.
```
Thanks for the review. I added some changes that address your suggestions

b-fein commented on docs/dev/setup/jenkins-gitlab.rst:
```diff
@@ -922,6 +924,58 @@ the following steps:
+The stable LTS versions can be viewed through the `changelog <https://www.jenkins.io/changelog-stable/>`__. and the corresponding
+Docker image can be found on `dockerhub <https://hub.docker.com/r/jenkins/jenkins/tags?page=1&ordering=last_updated>`__.
+
+1. Open the Jenkins Dockerfile and replace the value of ``FROM`` with ``jenkins/jenkins:lts``. This will use the latest LTS version.
```
```suggestion
1. Open the Jenkins Dockerfile and replace the value of ``FROM`` with ``jenkins/jenkins:lts``. After running the command ``docker pull jenkins/jenkins:lts``, this will use the latest LTS version in the following steps.
```

b-fein commented on docs/dev/setup/jenkins-gitlab.rst:
```diff
@@ -922,6 +924,58 @@ the following steps:
+Docker image can be found on `dockerhub <https://hub.docker.com/r/jenkins/jenkins/tags?page=1&ordering=last_updated>`__.
+
+1. Open the Jenkins Dockerfile and replace the value of ``FROM`` with ``jenkins/jenkins:lts``. This will use the latest LTS version.
+   You can also use specific LTS version.
```
```suggestion
   You can also use a specific LTS version.
```

b-fein commented on docs/dev/setup/jenkins-gitlab.rst:
```diff
@@ -922,6 +924,58 @@ the following steps:
+
+        docker-compose -f src/main/docker/gitlab-jenkins-mysql.yml up --build -d
+
+2. Build the new Docker image:
```
By introducing the alternative step above now the numbering is off by one.

b-fein reviewed (changes requested):
Looks good with the new additions since the last review. Only minor comments remaining.

I also noticed relevant changes further down in this file, line 1196 and following:
- The section ‘Changes’ should be one level lower (underlined  with ~~~).
- After that there already is a minimal ‘Upgrade Jenkins’ section. I think that one should be removed.

willergeorg reviewed (dismissed):
Could not find any additional typos.

NicolasUllmann reviewed (approved):
re-approve

ivanchimeno commented:
> Looks good with the new additions since the last review. Only minor comments remaining.
> 
> I also noticed relevant changes further down in this file, line 1196 and following:
> 
>     * The section ‘Changes’ should be one level lower (underlined  with ~~~).
> 
>     * After that there already is a minimal ‘Upgrade Jenkins’ section. I think that one should be removed.

Thanks for the review, I addressed the issues that you mentioned. What do you mean with section "Changes"? I can't seem to find it in the docs.

b-fein commented:
> What do you mean with section "Changes"? I can't seem to find it in the docs.
Oops. I meant ‘Caching’ now in line 1200 just above the old Upgrade Jenkins section

ivanchimeno commented:
> > What do you mean with section "Changes"? I can't seem to find it in the docs.
> 
> Oops, sorry. I meant ‘Caching’ now in line 1200 just above the old Upgrade Jenkins section

Ahh okay I made the changes :)

b-fein reviewed (approved):
Thank you! Looks good now.
